### PR TITLE
Stop survey api

### PR DESCRIPTION
**Pull request by harshilsharma63**

### CODE REVIEW

File: webapp/src/components/common/modal/modal.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (harshilsharma63) [inline webapp/src/components/common/modal/modal.tsx:1]: Note to reviewer - this is a copy of `GenericModal` from MM webapp, converted to a functional component and removed some unneeded props.

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/src/components/systemConsole/surveyResults/surveyResults.tsx [hunk 2]
```diff
@@ -35,7 +38,39 @@ function SurveyResults() {
         fetchSurveyStatus();
     }, [hydrateSurveyResults]);
 
-    const generateActions = (surveyResult: SurveyResult) => {
+    const handleStopSurvey = useCallback(async () => {
+        if (!currentSurveyID.current) {
+            return;
+        }
+
+        const updatedSurveyResult = [...surveyResults];
```
Comment (larkox) [inline webapp/src/components/systemConsole/surveyResults/surveyResults.tsx:46]: Nitpick: You don't need to create this list until you are going to set the state, so you can defer it until line 58. That way, if anything fails, you haven't created a new list "for nothing", which means less stress for the GC.

But yeah... probably not our bottleneck by far :stuck_out_tongue:

File: webapp/src/components/systemConsole/surveyResults/surveyResults.tsx [hunk 3]
```diff
@@ -119,6 +155,18 @@ function SurveyResults() {
                 </div>
             </div>
 
+            {
+                showConfirmationModal &&
+                <ConfirmationModal
+                    id='endSurveyConfirmationModal'
+                    title='Are you sure you want to end this survey?'
+                    bodyMessage='Reprehenderit aute laborum anim dolore magna mollit incididunt officia qui labore. Esse culpa ipsum ut esse fugiat do tempor duis mollit. Labore nisi nisi ut aliquip enim aute ex. Esse veniam proident id dolor laborum occaecat voluptate esse laborum cupidatat aliquip ut laborum id. Laborum nostrud officia excepteur. Consectetur ad nisi ullamco est cillum proident Lorem. Aliqua eu est magna deserunt est elit elit do.'
```
Comment (larkox) [inline webapp/src/components/systemConsole/surveyResults/surveyResults.tsx:163]: Do we want this Lorem ipsum to get to main?

**Review by larkox — APPROVED** [review]
LGTM, just to nits.

File: webapp/src/components/systemConsole/surveyResults/surveyResults.tsx (hunk 2 above)
Comment (harshilsharma63) [reply webapp/src/components/systemConsole/surveyResults/surveyResults.tsx:46]: Done! The GC thanks you for caring about it 🤣

File: webapp/src/components/systemConsole/surveyResults/surveyResults.tsx (hunk 3 above)
Comment (harshilsharma63) [reply webapp/src/components/systemConsole/surveyResults/surveyResults.tsx:163]: We'll update it with other text review. Its fine.

**Pull request merged**